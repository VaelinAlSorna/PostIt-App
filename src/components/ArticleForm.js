import React from 'react';
import moment from 'moment';

class ArticleForm extends React.Component {
    state = {
        title: '',
        text: '',
        createdAt: moment().format('D. M. Y')
    };
    onTitleChange = (e) => {
        const title = e.target.value;
        this.setState(() => ({
            title
        }))
    }
    onTextChange = (e) => {
        const text = e.target.value;
        this.setState(() => ({
            text
        }))
    }
    onSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit(this.state)
    }
    render () {
       return ( 
        <div> 
            <form onSubmit={this.onSubmit}> 
                <input 
                    onChange={this.onTitleChange} 
                    placeholder="Title" 
                    type="text" 
                    value={this.state.title}
                />
                <textarea 
                    onChange={this.onTextChange}
                    placeholder="Add text for your article"
                    value={this.state.text}
                >
                </textarea>
                <input type="submit" value="Submit" />
            </form>
        </div>
       );
    }
}

export default ArticleForm;