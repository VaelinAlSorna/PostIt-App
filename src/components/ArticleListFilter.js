import React from 'react';
import { connect } from 'react-redux';
import { setYourArticlesFilter, removeYourArticlesFilter, setTextFilter } from '../actions/filterActions';

export class ArticleListFilter extends React.Component {
    handleSeeAll = () => {
            this.props.removeYourArticlesFilter();
            this.props.setTextFilter('');
    }
    handleSeeYours = () => {
        this.props.setYourArticlesFilter();
    }
    disablingBtnSeeAll = () => {
        if(this.props.filter.yourArticles || !!this.props.filter.text) {
            return false;
        } else {
            return true;
        }
    }
    disablingBtnSeeYours = () => {
        if(this.props.user.name && !this.props.filter.yourArticles) {
            return false;
        } else {
            return true;
        }
    }
    onTextFilterChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }
    onSubmit = (e) => {
        //This is here in order to prevent user on mobile from sunmiting form
        e.preventDefault();
    }
    
    render () { 
        return  ( 
            <div className="filter">
                <form onSubmit={this.onSubmit} className="filter__group filter-form"> 
                    <input
                        className="filter-form__input"
                        onChange={this.onTextFilterChange} 
                        type="text" 
                        placeholder="Search Article By Name"
                        value={this.props.filter.text}
                    />
                </form>
                <button className="filter__group btn btn--green" onClick={this.handleSeeAll} disabled={this.disablingBtnSeeAll()}>See All Articles</button>
                <button className="filter__group btn btn--green" onClick={this.handleSeeYours} disabled={this.disablingBtnSeeYours()}>See Your Articles</button>
            </div>                
        )
    };
}

const mapStateToProps = (state) => ({
    filter: state.filter,
    user: state.auth
});

const mapDispatchToProps = (dispatch) => ({
    setYourArticlesFilter: () => dispatch(setYourArticlesFilter()),
    removeYourArticlesFilter: () => dispatch(removeYourArticlesFilter()),
    setTextFilter: (text) => dispatch(setTextFilter(text))
});

export default connect(mapStateToProps, mapDispatchToProps)(ArticleListFilter);

