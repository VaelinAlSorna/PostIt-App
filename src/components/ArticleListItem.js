import React from 'react';
//import { IoIosHeart, IoIosChatboxes } from "react-icons/io";
import { Link } from 'react-router-dom';


const ArticleListItem = ({ _id, title, createdAt, creator, comments, likes}) => (
    <Link to={`view/${_id}`}>
        <h3>{title}</h3>
        <p>Posted by: {creator} at: {createdAt}</p>
        {/*<p> <IoIosHeart /> {likes.length}</p>*/}
        {/*<p><IoIosChatboxes /> { comments.length }</p>*/}
    </Link>
);

export default ArticleListItem;