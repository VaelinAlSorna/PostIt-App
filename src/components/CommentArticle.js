import React from 'react';
import uuid from 'uuid';
import moment from 'moment';

class CommentArticle extends React.Component {
    state = {
        comment: '',
        createdAt: moment().format('D. M. Y'),
        error: ''
    };
    onTextareaChange = (e) => {
        const comment = e.target.value;
        this.setState(() => ({
            comment
        }));
    };
    onSubmit = (e) => {
        e.preventDefault();
        if (this.state.comment.trim().length < 3) {
            this.setState(() => ({
                error: 'Yout comment must contain at least 3 characters.'
            }));
        } else {
            const id = uuid();
            this.props.onSubmit({...this.state, id});
            this.setState(() =>  ({
                comment: '',
                error: ''
            }));
        }
    }
    render () {
        return (
            <form onSubmit={this.onSubmit}>
                { this.state.error && <p>{this.state.error}</p>}
                <textarea 
                    onChange={this.onTextareaChange} 
                    placeholder="What are your thoughts ?"
                    value={this.state.comment}
                >
                </textarea>
                <input type="submit" value="Comment" />
            </form>
        );
    };
};

export default CommentArticle;