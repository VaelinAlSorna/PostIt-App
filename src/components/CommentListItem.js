import React from 'react';

class CommentListItem extends React.Component {
   
    onDelete = () => {
        this.props.onClickDeleteComment(this.props.id);
    }
    render () { 
        return  (
                <li>
                    <p>{this.props.createdAt}</p>
                    <p>{this.props.comment}</p>
                    <button onClick={this.onDelete}>Delete</button>
                </li>
            )
    }
};

export default CommentListItem;