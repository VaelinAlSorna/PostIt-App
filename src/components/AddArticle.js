import React from 'react';
import { connect } from 'react-redux'
import ArticleForm from './ArticleForm';
import { startAddArticle } from '../actions/articlesActions';


export class AddArticle extends React.Component {
    onSubmit = (article) => {
        this.props.startAddArticle(article);
        this.props.history.push('/');
    };
    render () {
        return ( 
        <div>
            <h2>Creating Article</h2>
            <ArticleForm onSubmit={this.onSubmit} />
        </div>
        );
    };
};


const mapDispatchToProps = (dispatch) => ({
    startAddArticle: (article) => dispatch(startAddArticle(article))
});

export default connect( undefined, mapDispatchToProps)(AddArticle);

