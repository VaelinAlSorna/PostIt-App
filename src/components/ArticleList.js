import React from 'react';
import { connect } from 'react-redux';
import ArticleListItem from './ArticleListItem';
import ArticleListFilter from './ArticleListFilter';
import yourArticlesFilter from '../selectors/yourArticlesFilter';

export const ArticleList = (props) => (
    <div className="main-content">
        <ArticleListFilter />
        <ol className="list">
            {
                (props.articles.length === 0) 
                ? <p> No articles fulfill these search conditions. </p>
                : props.articles.map((article) => <ArticleListItem key={article._id} {...article} />).reverse()
            }
        </ol>
    </div>
);
 

const mapStateToProps = (state) => ({
    articles: yourArticlesFilter(state.articles, state.auth.id, state.filter),
});

export default connect(mapStateToProps)(ArticleList);