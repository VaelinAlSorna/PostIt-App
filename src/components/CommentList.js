import React from 'react';
import CommentListItem from './CommentListItem';

const CommentList = (props) => (
         <ol>
            {
                props.comments.map((comment) => <CommentListItem 
                                                userId={props.userId}
                                                onClickDeleteComment={props.onClickDeleteComment} 
                                                key={comment._id} 
                                                {...comment}/>
                                            )
            }
        </ol>
)

export default CommentList;