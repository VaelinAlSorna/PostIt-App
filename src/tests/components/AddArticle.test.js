import React from 'react';
import { shallow } from 'enzyme';
import { AddArticle } from '../../components/AddArticle';
import articles from '../fixtures/articlesFixtures';

let startAddArticle, history, wrapper;

beforeEach(() => {
    startAddArticle = jest.fn();
    history = { push: jest.fn()};
    wrapper = shallow(<AddArticle startAddArticle={startAddArticle} history={history}/>)
});

test('should render AddArticle correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit call', () => {
    wrapper.find('ArticleForm').prop('onSubmit')(articles[0]);
    expect(startAddArticle).toHaveBeenLastCalledWith(articles[0]);
    expect(history.push).toHaveBeenLastCalledWith('/');
});