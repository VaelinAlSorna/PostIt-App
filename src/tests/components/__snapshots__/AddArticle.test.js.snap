// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddArticle correctly 1`] = `
<div>
  <h2>
    Creating Article
  </h2>
  <ArticleForm
    onSubmit={[Function]}
  />
</div>
`;
